lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc

```diff
@@ -324,6 +324,10 @@ function dosomething_reportback_form_validate($form, &$form_state) {
 function dosomething_reportback_form_submit($form, &$form_state) {
   $values = $form_state['values'];
 
+  $campaign = node_load($values['nid']);
+  $language_code = $language_code = $campaign->language;
+  $values['run_nid'] = $campaign->field_current_run[$language_code][0]['target_id'];
```

Comment:
There will be a notice here if `$campaign->field_current_run` doesn't have a `$language_code` key. Please handle this case properly.